webapp/tests/karma/ts/services/uhc-stats.service.spec.ts

```diff
@@ -0,0 +1,286 @@
+import { TestBed } from '@angular/core/testing';
+import sinon from 'sinon';
+import { expect } from 'chai';
+import * as moment from 'moment';
+
+import { UHCStatsService } from '@mm-services/uhc-stats.service';
+import { DbService } from '@mm-services/db.service';
+import { ContactTypesService } from '@mm-services/contact-types.service';
+import { SessionService } from '@mm-services/session.service';
+import { AuthService } from '@mm-services/auth.service';
+
+describe('UHCStats Service', () => {
+  let service: UHCStatsService;
+  let clock;
+  let localDb;
+  let dbService;
+  let contactTypesService;
+  let sessionService;
+  let authService;
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers(moment('2021-04-07 18:18:18').valueOf());
+    localDb = { query: sinon.stub() };
+    dbService = { get: sinon.stub().returns(localDb) };
+    sessionService = { isDbAdmin: sinon.stub() };
+    authService = { has: sinon.stub() };
+    contactTypesService = {
+      getTypeId: sinon.stub(),
+      get: sinon.stub()
+    };
+
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: DbService, useValue: dbService },
+        { provide: ContactTypesService, useValue: contactTypesService },
+        { provide: SessionService, useValue: sessionService },
+        { provide: AuthService, useValue: authService },
+      ]
+    });
+
+    service = TestBed.inject(UHCStatsService);
+  });
+
+  afterEach(() => {
+    clock.restore();
+    sinon.restore();
+  });
+
+  it('should return UHC Interval', () => {
+    const visitCountSettings = {
+      monthStartDate: 26,
+      visitCountGoal: 5
+    };
+
+    const result = service.getUHCInterval(visitCountSettings);
+
+    expect(result).to.deep.equal({
+      start: moment('2021-03-26 00:00:00.000').valueOf(),
+      end: moment('2021-04-25 23:59:59.999').valueOf()
+    });
+  });
+
+  it('should not return UHC Interval if setting arent provided', () => {
+    const result = service.getUHCInterval(null);
+
+    expect(result).to.be.undefined;
+  });
+
+  it('should get home visit stats', async () => {
```

Comment:
Please add an additional test that checks that visits from the same day are not counted.

Reply:
Done!